daml-assistant/integration-tests/src/Main.hs

```diff
@@ -231,11 +231,12 @@ packagingTests tmpDir = testGroup "packaging"
     , testCaseSteps "Build migration package" $ \step -> do
         let projectA = tmpDir </> "a"
         let projectB = tmpDir </> "b"
-        let projectMigrate = tmpDir </> "migrateAB"
-        let aDar = projectA </> distDir </> "a.dar"
-        let bDar = projectB </> distDir </> "b.dar"
-        let bUpgradedDar = tmpDir </> "b_upgraded.dar"
-        step "Creating project a..."
+        let projectUpgrade = tmpDir </> "upgrade"
+        let aDar = projectA </> distDir </> "a-1.0.dar"
```

Comment:
What happens if I pass in two dars with the same version but different contents? Do I get a sensible error message or do things blow up badly? Might also makes sense to add a test for that.

Reply:
Compilation of the migration code would fail, package names need to be different, otherwise you will get overlapping instances.